junit-vintage-engine/src/main/java/org/junit/vintage/engine/execution/TestRun.java

```diff
@@ -95,7 +95,12 @@ boolean isDescendantOfRunnerTestDescriptor(TestDescriptor testDescriptor) {
 		}
 		// @formatter:off
 		return descriptors.stream()
-				.filter(testDescriptor -> description == testDescriptor.getDescription())
+				.filter(testDescriptor -> {
+					// @marcphillip: why are Descriptions equated by reference and not using .equals()?
+					@SuppressWarnings("ReferenceEquality")
+					boolean result = (description == testDescriptor.getDescription());
+					return result;
+				})
```

Comment:
Okay, thanks for the explanation @marcphilipp!

I'll remove the question-comment above now. :)